app/models/metric/targets.rb

```diff
@@ -9,22 +9,20 @@ def self.perf_capture_always=(options)
   end
 
   def self.capture_infra_targets(zone, options)
-    # Preload all of the objects we are going to be inspecting.
-    includes = {:ext_management_systems => {:hosts => {:ems_cluster => :tags, :tags => {}}}}
-    includes[:ext_management_systems][:hosts][:storages] = :tags unless options[:exclude_storages]
-    MiqPreloader.preload(zone, includes)
+    load_infra_targets_data(zone, options)
+    all_hosts = capture_host_targets(zone)
+    targets = hosts = only_enabled(all_hosts)
+    targets += capture_storage_targets(all_hosts) unless options[:exclude_storages]
+    targets += capture_vm_targets(hosts) unless options[:exclude_vms]
```

Comment:
So, I was confused by the `all_hosts` and `hosts` here.  Note, the latter is "enabled_hosts".  So, we capture_storage_targets using all_hosts but we capture vm_targets with just enabled hosts.  But then...capture_vm_targets does

```ruby
hosts.select(&:perf_capture_enabled?)
```

Is this check already done (redundant) when we captured the `host = only_enabled(all_hosts)`?